sklearn/preprocessing/_polynomial.py

```diff
@@ -1150,16 +1220,19 @@ def transform(self, X):
                 )
             XBS = sparse.hstack(output_list, format="csr")
         elif self.sparse_output:
-            # TODO: Remove ones scipy 1.10 is the minimum version. See comments above.
+            # TODO: Remove once scipy 1.10 is the minimum version. See comments above.
             XBS = sparse.csr_matrix(XBS)
 
+        # replace any np.nan values with 0
+        XBS = np.nan_to_num(XBS, nan=0)
```

Comment:
Then instead of relying of the location nans in the XBS, use the missingness information extracted from `X` at the beginning of `transform` to overwrite the XBS values for the matching rows and columns by meaningful 0.0 values.